### PR TITLE
For #81 - fix undefined variable in get_type_by_name()

### PR DESCRIPTION
**Pull request by arporter**
Small code fix and added new test for the associated routine.

### CODE REVIEW

**Comment by TeranIvy** [comment]
Happy to review it.

File: src/fparser/one/block_statements.py [hunk 1]
```diff
@@ -100,6 +101,16 @@ class HasImplicitStmt(object):
     a = AttributeHolder(implicit_rules={})
 
     def get_type_by_name(self, name):
+        '''
+        Returns an object of the correct type (Integer or Real) using
+        Fortran's implicit typing rules for the supplied variable name.
+
+        :param str name: the variable name
```
[the remark this answers is not in the document]
Comment (arporter) [reply src/fparser/one/block_statements.py]: Good point, done.

**Comment by arporter** [comment]
I've updated for pylint as requested and added further tests to hit the missed lines. They then revealed a Python 3 problem which I have also fixed.

**Comment by arporter** [comment]
I've rebased this branch onto the head of master. Ready for second review.

**Comment by arporter** [comment]
(Forgot to say that I haven't updated the class attribute named "a" because, even though it's a rubbish name, it permeates a lot of the code. Something for a separate issue.)

**Pull request merged**